#5999 - remove unnecessary join to cases

BarnaBartha opened the pull request:
<!--
If you've never submitted a pull request to the SORMAS repository before or this is your first time using this template, please read the Contributing guidelines (https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/CONTRIBUTING.md) for an explanation of our guidelines regarding pull requests. You don't have to remove this comment or from your pull request as it will automatically be hidden.

Please specify the number of the issue this pull request is related to after the #.
-->
Fixes #5999

StefanKock reviewed (commented):
Two questions come to my mind:

1. Is the default method `personService.createUserFilter(cb, cq, person)` used somewhere else also and now unimplemented? If it's not supposed to be used, it should throw an `UnsupportedOperationException` or similar.
2. How did the generated SQL change in a representative unit test?

BarnaBartha commented:
> 2. How did the generated SQL change in a representative unit test?

`select distinct person0_.uuid                                           as col_0_0_,
                person0_.firstName                                      as col_1_0_,
                person0_.lastName                                       as col_2_0_,
                person0_.approximateAge                                 as col_3_0_,
                person0_.approximateAgeType                             as col_4_0_,
                person0_.birthdate_dd                                   as col_5_0_,
                person0_.birthdate_mm                                   as col_6_0_,
                person0_.birthdate_yyyy                                 as col_7_0_,
                person0_.sex                                            as col_8_0_,
                district2_.name                                         as col_9_0_,
                location1_.street                                       as col_10_0_,
                location1_.houseNumber                                  as col_11_0_,
                location1_.postalCode                                   as col_12_0_,
                location1_.city                                         as col_13_0_,
                (select personcont16_.contactInformation
                 from PersonContactDetail personcont16_
                 where personcont16_.person_id = person0_.id
                   and personcont16_.primaryContact = 1
                   and personcont16_.personContactDetailType = 'PHONE') as col_14_0_,
                (select personcont17_.contactInformation
                 from PersonContactDetail personcont17_
                 where personcont17_.person_id = person0_.id
                   and personcont17_.primaryContact = 1
                   and personcont17_.personContactDetailType = 'EMAIL') as col_15_0_,
                person0_.changeDate                                     as col_16_0_,
                case
                    when 1 = 1 or (user4_.id is not null) and user4_.uuid = 'SCJTTT-LXXH7W-CTGWSP-IM4X2HVI' or 1 = 1 or
                         (user7_.id is not null) and user7_.uuid = 'SCJTTT-LXXH7W-CTGWSP-IM4X2HVI' or 1 = 1 or 1 = 1 or
                         (user10_.id is not null) and user10_.uuid = 'SCJTTT-LXXH7W-CTGWSP-IM4X2HVI' or 1 = 1 then 1
                    else 0 end                                          as col_17_0_
from Person person0_
         left outer join Location location1_ on person0_.address_id = location1_.id
         left outer join District district2_ on location1_.district_id = district2_.id
         left outer join cases cases3_ on person0_.id = cases3_.person_id
         left outer join users user4_ on cases3_.reportingUser_id = user4_.id
         left outer join Contact contacts5_ on person0_.id = contacts5_.person_id
         left outer join cases case6_ on contacts5_.caze_id = case6_.id
         left outer join users user7_ on contacts5_.reportingUser_id = user7_.id
         left outer join cases case8_ on contacts5_.caze_id = case8_.id
         left outer join EventParticipant eventparti9_ on person0_.id = eventparti9_.person_id
         left outer join users user10_ on eventparti9_.reportingUser_id = user10_.id
         left outer join events event11_ on eventparti9_.event_id = event11_.id
         left outer join Location location12_ on person0_.address_id = location12_.id
         left outer join Region region13_ on location12_.region_id = region13_.id
         left outer join District district14_ on location12_.district_id = district14_.id
         left outer join Community community15_ on location12_.community_id = community15_.id
where (cases3_.deleted = 0 or contacts5_.deleted = 0 or eventparti9_.deleted = 0)
  and person0_.presentCondition = 1
order by person0_.changeDate desc`

StefanKock commented:
Rebased branch to include fixed unit tests for #6035
